sidm/tools/sidm_processor.py

```diff
@@ -211,26 +208,20 @@ def build_lepton_jets(self, objs, lj_reco):
                 with_name="LorentzVector",
                 behavior=cvec.behavior
             )
-
-            # add jet constituent info
-            const_vec = ak.zip(
-                {"x": cluster.constituents().x,
-                 "y": cluster.constituents().y,
-                 "z": cluster.constituents().z,
-                 "t": cluster.constituents().t,
-                 "charge": cluster.constituents().charge,
-                  "part_type":cluster.constituents().part_type},
-                 with_name="LorentzVector",
-                 behavior=cvec.behavior)
-
+            forms = {f : cluster.constituents()[f] for f in cluster.constituents().fields}
+            forms["x"] = cluster.constituents().x
```

Comment:
Is it necessary to handle x, y, z, t separately? If not doing this gives errors for unknown reasons, it might be good to add a comment saying so.